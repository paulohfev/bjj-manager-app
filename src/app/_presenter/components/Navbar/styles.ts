import { Theme } from '@mui/material'

const styles = {
  appBar: (theme: Theme) => ({
    display: 'flex',
    height: theme.appBar.height,
    justifyContent: 'center',
    padding: '0 24px',
  }),
}

export default styles
