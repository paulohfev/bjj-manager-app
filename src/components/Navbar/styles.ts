import { Theme } from '@mui/material'

const styles = {
  appBar: (theme: Theme) => ({
    display: 'flex',
    height: theme.appBar.height,
    justifyContent: 'center',
    padding: '0 24px',
  }),

  container: {
    alignItems: 'center',
    display: 'flex',
    justifyContent: 'space-between',
    width: '100%',
  },
}

export default styles
