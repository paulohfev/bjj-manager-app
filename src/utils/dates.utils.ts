import { DateFormat } from '@/constants/dateFormats'

import dayjs from '../dates'

export const formatDate = (date: string): string => {
  return dayjs(date).format(DateFormat.DD_MM_YYYY_slash)
}
