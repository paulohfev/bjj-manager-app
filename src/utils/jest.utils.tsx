import React from 'react'

import { SessionProvider } from 'next-auth/react'

import { ThemeOptions, ThemeProvider, createTheme } from '@mui/material'

import { render } from '@testing-library/react'

import muiThemeConfig from '@/theme/mui-theme-config'

export const renderWithTheme = (component: React.ReactNode) => {
  const mockExpirationDate = new Date(Date.now() + 60 * 60 * 1000).toISOString()

  return render(
    <SessionProvider
      session={{
        user: {
          email: 'mary@test.com',
        },
        expires: mockExpirationDate,
      }}
    >
      <ThemeProvider theme={createTheme(muiThemeConfig as ThemeOptions)}>{component}</ThemeProvider>
    </SessionProvider>
  )
}
