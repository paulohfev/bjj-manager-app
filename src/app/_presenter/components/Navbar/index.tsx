'use client'

import { AppBar, Typography } from '@mui/material'

import styles from './styles'

const Navbar: React.FC = () => {
  return (
    <AppBar position='fixed' sx={styles.appBar}>
      <Typography>Navbar</Typography>
    </AppBar>
  )
}

export default Navbar
