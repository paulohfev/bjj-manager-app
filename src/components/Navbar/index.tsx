'use client'

import { AppBar, Box, Typography } from '@mui/material'

import AuthControls from '../AuthControls'
import styles from './styles'

const Navbar: React.FC = () => {
  return (
    <AppBar position='fixed' sx={styles.appBar}>
      <Box sx={styles.container}>
        <Typography>Navbar</Typography>

        <AuthControls />
      </Box>
    </AppBar>
  )
}

export default Navbar
