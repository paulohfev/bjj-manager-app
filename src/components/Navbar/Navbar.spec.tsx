import React from 'react'

import '@testing-library/jest-dom'

import { renderWithTheme } from '@/utils/jest.utils'

import Navbar from '.'

describe('Navbar', () => {
  it('should match the snapshot', () => {
    const button = renderWithTheme(<Navbar />)
    expect(button).toMatchSnapshot()
  })
})
