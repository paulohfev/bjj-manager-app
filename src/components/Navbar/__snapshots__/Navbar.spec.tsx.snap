// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-jrhfyx-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiBox-root css-1hl232g"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Navbar
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1tmrq3e-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Sign out
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-jrhfyx-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiBox-root css-1hl232g"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Navbar
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1tmrq3e-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Sign out
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
